modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/test/java/com/liferay/portal/search/elasticsearch7/internal/connection/ProxyConfigTest.java

```diff
@@ -53,28 +53,54 @@ public void tearDown() {
 	}
 
 	@Test
-	public void testShouldApplyConfigIfHttpHasProxyConfig() {
-		Mockito.when(
-			_http.hasProxyConfig()
-		).thenReturn(
-			true
-		);
-
+	public void testShouldApplyConfigWithHostAndPort() {
 		ProxyConfig.Builder builder = ProxyConfig.builder(_http);
 
 		ProxyConfig proxyConfig = builder.host(
 			"http://proxy"
+		).port(
+			32000
 		).build();
 
 		Assert.assertTrue(proxyConfig.shouldApplyConfig());
 	}
 
 	@Test
-	public void testShouldApplyConfigWithHostAndPort() {
+	public void testShouldApplyConfigWithHostAndPortInSystemProperties() {
+		System.setProperty("http.proxyHost", "http://proxy");
+		System.setProperty("http.proxyPort", "32000");
+
 		ProxyConfig.Builder builder = ProxyConfig.builder(_http);
 
+		ProxyConfig proxyConfig = builder.build();
+
+		Assert.assertTrue(proxyConfig.shouldApplyConfig());
+	}
+
+	@Test
+	public void testShouldApplyConfigWithHostAndPortOfProxyHost() {
+		ProxyConfig.Builder builder = ProxyConfig.builder(_http);
+
+		String differentDomain = "diffdomain";
+		String differentNetworkAddress = "http://diffdomain:9200";
+		String domain = "domain";
+
+		Mockito.when(
+			_http.getDomain(differentNetworkAddress)
+		).thenReturn(
+			differentDomain
+		);
+
+		Mockito.when(
+			_http.isNonProxyHost(differentDomain)
+		).thenReturn(
+			domain.equals(differentDomain)
+		);
+
 		ProxyConfig proxyConfig = builder.host(
 			"http://proxy"
```

Comment:
@joshuacords Shouldn't we set the proxy `host` here to `domain`?

Reply:
Asking, because if I understand the test case well, `domain` is a proxy host and `diffdomain` is a non-proxy host.

Reply:
@lipusz, this case is testing `domain` being set as a nonProxyHost but `diffdomain` as what's being set as the host for the connection. `diffdomain` will be used because it's not excluded as a nonProxyHost. `testShouldNotApplyConfigWithHostAndPortOfNonProxyHost` tests when a `domain` of the host connection is the same as a value set as a nonProxyHost.

Reply:
Some renaming would help. Thanks!

Reply:
@lipusz Do you want me to force push that change?

Reply:
Sure, go ahead.

Reply:
And trigger forward if tests look good.